runtime/src/ancient_append_vecs.rs

```diff
@@ -39,7 +38,7 @@ impl<'a> AccountsToStore<'a> {
     /// available_bytes: how many bytes remain in the primary storage. Excess accounts will be directed to an overflow storage
     pub fn new(
         mut available_bytes: u64,
-        stored_accounts: &'a HashMap<Pubkey, FoundStoredAccount>,
+        stored_accounts: &'a Vec<(&'a Pubkey, &'a FoundStoredAccount<'a>)>,
```

Comment:
My stable clippy thinks this ought to be `&'a [(&'a Pubkey, &'a FoundStoredAccount<'a>)]`.
Is it right? Not sure why nightly (CI) clippy didn't agree